### DISCUSSION TITLE
Do not create new record, if related record exists

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#16562] [General]
Create entity workflow have this check box "Do not create new record, if related record exists" which actually doesn't work at all I think (latest stable).
I assume it should disallow creating new related record (new related entity), but if it works I wonder how it should.
1. Do not create new record, if related record exists - per user? (the most logical)
2. Do not create new record, if related record exists - applying to all users

Maybe here is something I don't understand? Can I have a help with this little thing?

Best

**Comment by rskrzypczak** [comment]
https://github.com/YetiForceCompany/YetiForceCRM/blob/b94aa95ef4014503264da69e5e399a46297309c1/modules/com_vtiger_workflow/tasks/VTCreateEntityTask.php#L99-L102

As you can see, this only works in the 'mapping form panel' option and verifies that there is a related record available to the user calling this action.
The functionality should actually be improved, but this has a low priority for us.
Feel free to make improvements yourself and push a pull request :)

**Comment by s2b-git** [comment]
I understand, but it not works anyway :-) I wish to recommend improvement, or attempt to  fix bug, but firstly I don't really understand what it suppose to do; what is purpose of this check-box.
Currently, if I check this box, as the same user I can keep creating related entities endlessly..so i think it is bug, or maybe I missed something


Best regards
